staging/src/k8s.io/apimachinery/pkg/api/meta/interfaces.go

```diff
@@ -133,11 +135,112 @@ type RESTMapper interface {
 	ResourceSingularizer(resource string) (singular string, err error)
 }
 
+// RESTMapperWithContext allows clients to map resources to kind, and map kind and version
+// to interfaces for manipulating those objects. It is primarily intended for
+// consumers of Kubernetes compatible REST APIs as defined in docs/devel/api-conventions.md.
+//
+// The Kubernetes API provides versioned resources and object kinds which are scoped
+// to API groups. In other words, kinds and resources should not be assumed to be
+// unique across groups.
+type RESTMapperWithContext interface {
+	// KindFor takes a partial resource and returns the single match.  Returns an error if there are multiple matches
+	KindForWithContext(ctx context.Context, resource schema.GroupVersionResource) (schema.GroupVersionKind, error)
```

Comment:
I am using the `*WithContext` pattern consistently because that's what we settled on in Kubernetes. However, it leads to long names which in this case are also a bit odd.

In Go, `*Context` is used more often than `*WithContext`:
```
# e.g. database/sql/sql.go:func (db *DB) PingContext(ctx context.Context) error {
$ grep -r 'func.*Context(ctx' /nvme/gopath/go/src/ | grep -v WithContext | wc -l
55
# e.g. net/http/request.go:func NewRequestWithContext(ctx context.Context, method, url string, body io.Reader) (*Request, error)
$ grep -r 'func.*WithContext(ctx' /nvme/gopath/go/src/ | wc -l
5
```

Should we follow that practice and use the shorter `*Context` pattern?

Then this becomes:
```golang
type RESTMapperContext interface {
    KindForContext(ctx context.Context, resource schema.GroupVersionResource) (schema.GroupVersionKind, error)
```

Reply:
Although `*Context` is used far more and makes sense, `RESTMapperContext` may mislead people about it behaves as a Context rather than having a Context.